docs/templates.md

```diff
@@ -0,0 +1,175 @@
+# Templates
+
+Sydent uses parametrised templates to generate the content of emails it
+sends and webpages it shows to users. Example templates can be found in
+the [res](https://github.com/matrix-org/sydent/tree/main/res) folder.
+
+## Branding
+
+Sending a value for `brand` to some API endpoints allows for different
+email and http templates to be used. These templates should be stored
+in a file structure like this:
+
+```
+ root_template_dir/
+     brand1/
+         [ brand1 template files ]
+     brand2/
+         [ brand2 template files ]
+```
+
+The config option `templates.root_directory` should be set to the path
+of `root_template_dir` and `templates.default_brand` should be set to
+the sub directory to use if no `brand` (or an invalid `brand`) is provided
+by the request.
+
+Here are the requests that can contain a value for `brand` and what
+template the brand value is used to select:
+
+Invite email templates:
+ - `POST /_matrix/identity/v2/validate/msisdn/requestToken`
+
+Verification email templates:
+ - `POST /_matrix/identity/v2/validate/email/requestToken`
+
+Verification SMS templates - BRAND CURRENTLY IGNORED:
+ - `POST /_matrix/identity/v2/validate/msisdn/requestToken`
+
+Verification response templates:
+ - `GET /_matrix/identity/v2/validate/email/submitToken`
+ - `GET /_matrix/identity/v2/validate/msisdn/submitToken`
+
+
+## Template formats
+
+### `.eml.j2` files
+
+Files ending in ".eml.j2" are Jinja templates. 
+
+Using the `urlencode` Jinja filter encodes the contents for URLs suitably.
+All variables are automatically made safe for HTML.
+
+(If needed the `safe` filter can be used to prevent the HTML encoding, i.e.
+declare the variable's contents "safe")
+
+See the Jinja [documentation](https://jinja.palletsprojects.com/en/3.0.x/templates/#variables)
+for more instructions on how to write these templates.
+
+### DEPRECATED `.eml` template files 
+
+Files ending in ".eml" can use python `%(variable)s` string substitution.
+
+Appending "_forurl" or "_forhtml" to any of the variable names listed below
+returns their values encoded suitably for URLs or HTML respectively.
+
+For example ">" in the HTML encoded version would be replaced with "&gt".
```

Comment:
I'm not sure what this line is referring to?

It might be worth mentioning that you need to double & to get a raw one. E.g. `&&` -> `&` after processing.

Reply:
What do you mean by referring to? 

By "listed below" I meant the tables of variables and line 65 was referring to the "_forhtml" suffix mentiond on line 62 if that answers your question?

Reply:
Maybe this is clearer?

```suggestion
For example ">" in the `_forhtml` version would be replaced with "&gt".
```